src/ast/spans.rs

```diff
@@ -777,11 +778,9 @@ impl Spanned for ConditionalStatements {
             ConditionalStatements::Sequence { statements } => {
                 union_spans(statements.iter().map(|s| s.span()))
             }
-            ConditionalStatements::BeginEnd {
-                begin_token: AttachedToken(start),
-                statements: _,
-                end_token: AttachedToken(end),
-            } => union_spans([start.span, end.span].into_iter()),
+            ConditionalStatements::BeginEnd(bes) => {
+                union_spans([bes.begin_token.0.span, bes.end_token.0.span].into_iter())
```

Comment:
Can we `impl Spanned for BeginEndStatements` and then call `bes.span()` here instead? That would make the span impl reusable by nother nodes that embed the `BeginEndStatements`

Reply:
I think I understand what you're asking for, I'll give the implementation a shot

Reply:
Done 👍